sklearn/decomposition/tests/test_online_lda.py

```diff
@@ -0,0 +1,292 @@
+
+
+def test_lda_fit_batch():
+    """
+    Test LDA batch learning_offset (`fit` method with 'batch' learning)
+    """
+    rng = np.random.RandomState(0)
+    n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior,
+                                    learning_method='batch', random_state=rng)
+    lda.fit(X)
+
+    correct_idx_grps = [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
+    for c in lda.components_:
+        top_idx = set(c.argsort()[-3:][::-1])
+        assert_true(tuple(sorted(top_idx)) in correct_idx_grps)
+
+
+def test_lda_fit_online():
+    """
+    Test LDA online learning (`fit` method with 'online' learning)
+    """
+    rng = np.random.RandomState(0)
+    n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior, learning_offset=30.,
+                                    learning_method='online', random_state=rng)
+    lda.fit(X)
+
+    correct_idx_grps = [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
+    for c in lda.components_:
+        top_idx = set(c.argsort()[-3:][::-1])
+        assert_true(tuple(sorted(top_idx)) in correct_idx_grps)
+
+
+def test_lda_partial_fit():
+    """
+    Test LDA online learning (`partial_fit` method)
+    (same as test_lda_batch)
+    """
+    rng = np.random.RandomState(0)
+    n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior, learning_offset=30.,
+                                    random_state=rng)
+    for i in xrange(3):
+        lda.partial_fit(X)
+
+    correct_idx_grps = [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
+    for c in lda.components_:
+        top_idx = set(c.argsort()[-3:][::-1])
+        assert_true(tuple(sorted(top_idx)) in correct_idx_grps)
+
+
+def test_lda_dense_input():
+    """
+    Test LDA with dense input.
+    """
+    rng = np.random.RandomState(0)
+    X = rng.randint(5, size=(20, 10))
+    n_topics = 3
+    doc_topic_prior = 1. / n_topics
+    topic_word_prior = 1. / n_topics
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior, random_state=rng)
+    X_trans = lda.fit_transform(X)
+    assert_true((X_trans > 0.0).any())
+
+
+def test_lda_fit_transform():
+    """
+    Test LDA fit_transform & transform
+    fit_transform and transform result should be the same
+    """
+    for method in ('online', 'batch'):
+        rng = np.random.RandomState(0)
+        n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
```

Comment:
I feel that this data is a bit too trivial for this check... I don't have a great alternative, though.